### PR TITLE
Cassandra 18508

### PR DESCRIPTION
**Pull request by maulin-vasavada**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

File: src/java/org/apache/cassandra/config/DatabaseDescriptor.java [hunk 1]
```diff
@@ -1251,7 +1254,9 @@ public static void applySslContext()
         {
             SSLFactory.validateSslContext("Internode messaging", conf.server_encryption_options, REQUIRED, true);
             SSLFactory.validateSslContext("Native transport", conf.client_encryption_options, conf.client_encryption_options.getClientAuth(), true);
+            SSLFactory.validateSslContext("JMX transport", conf.jmx_encryption_options, conf.jmx_encryption_options.getClientAuth(), true);
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/config/DatabaseDescriptor.java:1257]: @maulin-vasavada how do you know that `conf.jmx_encryption_options` is not `null` here?

**Comment by smiklosovic** [comment]
@maulin-vasavada please update `cassandra.yaml` with this new section as well. (probably add it to `cassandra_latest.yaml` too and update `test/conf/latest_diff.yaml`)

**Comment by maulin-vasavada** [comment]
Closing since the PR is opened/reviewed on the Apache/Cassandra https://github.com/apache/cassandra/pull/3638

**Comment by maulin-vasavada** [comment]
Closing since the PR is opened/reviewed on the Apache/Cassandra https://github.com/apache/cassandra/pull/3638

**Pull request closed without merging**